JS/Lesson34/onder/my-app/src/components/NewRecipe/index.jsx

```diff
@@ -1,77 +1,139 @@
+import React, { useState } from 'react';
 import { useForm } from 'react-hook-form';
-import './styles.css';
 
-// const useMyHook = () => {
-//   const myData = {id: 'xxxx', name:'Data name'};
-//   const deleteMyData = () => {console.log('Deleting data...')};
+export const NewRecipe = () => {
+  const {
+    register,
+    handleSubmit,
+    watch,
+    setValue,
+    formState: { errors },
+  } = useForm();
 
-//   return [ myData, deleteMyData ];
-// }
+  const [formData, setFormData] = useState(recipeData);
+
+  const strCategory = watch('strCategory');
+
+  Object.keys(formData).forEach((key) => {
+    setValue(key, formData[key]);
+  });
+
+  const onSubmit = (data) => {
+    Object.keys(data).forEach((key) => {
```

Comment:
After looking at the entire code of this component, I'd recommend to remove the state (`const [formData, setFormData] = useState({});`) completely as it's not used anywhere. You only try to update it but never use it.

Normally, we'd either send an API request here or update application storage but we'll do it in the following lessons.

Since you use react-hook-form, you get `data` parameter in the `onSubmit` function that contain all the data that you need. 
What you need to do before using this data is to trim it as it may have spaces (e.g. if someone types '    Pizza   ' instead of 'Pizza').